### PR TITLE
docs(sematext_logs, sematext_metrics sink): Use generated docs, unify region behavior

### LINKED ISSUE #16337: Convert the `sematext_metrics` sink to use auto-generated docs

### LINKED ISSUE #16348: Convert the `sematext_logs` sink to use auto-generated docs

### PR DESCRIPTION
**Pull request by spencergilbert**
Closes #16348
Closes #16337

This also updates how `region` is handled across both of the sinks, unifying them in using `region` as a non-optional value. As I understand the code this should be backward compatible.

- `sematext_logs` defaulted to `us` inside the code itself, now it defaults to it in the `serde` attribute.
- `sematext_metrics` required a `region` to be set, so if previously used the new `serde` default would be overwritten.

### CODE REVIEW

File: src/sinks/sematext/mod.rs [hunk 1]
```diff
@@ -5,14 +5,18 @@ pub use self::{logs::SematextLogsConfig, metrics::SematextMetricsConfig};
...
-    /// US region.
+    /// United States
     Us,
 
-    /// EU region.
+    /// Europe
     Eu,
 }
+
+fn default_region() -> Region {
```
Comment (spencergilbert) [inline src/sinks/sematext/mod.rs:20]:
```suggestion
const fn default_region() -> Region {
```

File: src/sinks/sematext/logs.rs [hunk 2]
```diff
@@ -24,14 +24,21 @@ use crate::{
...
 #[derive(Clone, Debug)]
 pub struct SematextLogsConfig {
+    #[serde(default = "super::default_region")]
     #[configurable(derived)]
-    region: Option<Region>,
+    region: Region,
 
     /// The endpoint to send data to.
+    ///
+    /// Setting this option overrides the `region`.
```
Comment (neuronull) [inline src/sinks/sematext/logs.rs:33]: nit nit nit: "This option overrides `region`." , or "Setting this option overrides the `region` option" is more clear imo. I find it slightly confusing wording to use "setting" and "option" in the same sentence when "option" could be interchanged with "setting" 🙄

**Review by neuronull — APPROVED** [review]

File: src/sinks/sematext/logs.rs (hunk 2 above)
Comment (spencergilbert) [reply src/sinks/sematext/logs.rs]: Resolved in [cd5b1b8](https://github.com/vectordotdev/vector/pull/16405/commits/cd5b1b805ded4ca84971e261e3c7f442970f3a2d)

Definitely see what you're saying with `option` vs `setting` - I'll think about that.

File: src/sinks/sematext/logs.rs [hunk 3]
```diff
@@ -30,7 +30,7 @@ pub struct SematextLogsConfig {
 
     /// The endpoint to send data to.
     ///
-    /// Setting this option overrides the `region`.
+    /// Setting this option overrides the `region` option.
```
Comment (neuronull) [inline src/sinks/sematext/logs.rs:33]: Yeah with this now, I'd almost just do

"This option overrides the `region` option."

**Pull request merged**